builtin/credential/appgroup/path_login.go

```diff
@@ -0,0 +1,88 @@
+package appgroup
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/vault/logical"
+	"github.com/hashicorp/vault/logical/framework"
+)
+
+func pathLogin(b *backend) *framework.Path {
+	return &framework.Path{
+		Pattern: "login$",
+		Fields: map[string]*framework.FieldSchema{
+			"selector": &framework.FieldSchema{
+				Type:        framework.TypeString,
+				Description: "Identifier of the category the UserID belongs to.",
+			},
+			"user_id": &framework.FieldSchema{
+				Type:        framework.TypeString,
+				Default:     "",
+				Description: "UserID of the App.",
+			},
+		},
+		Callbacks: map[logical.Operation]framework.OperationFunc{
+			logical.UpdateOperation: b.pathLoginUpdate,
+		},
+		HelpSynopsis:    pathLoginHelpSys,
+		HelpDescription: pathLoginHelpDesc,
+	}
+}
+
+func (b *backend) pathLoginRenew(req *logical.Request, data *framework.FieldData) (*logical.Response, error) {
+	selectorType := req.Auth.InternalData["selector_type"].(string)
+	selectorValue := req.Auth.InternalData["selector_value"].(string)
+	if selectorType == "" || selectorValue == "" {
+		return nil, fmt.Errorf("failed to fetch selector type and/or selector value during renewal")
+	}
+	resp, err := b.validateSelector(req.Storage, selectorType, selectorValue)
+	if err != nil {
+		return nil, fmt.Errorf("failed to validate selector during renewal:%s", err)
+	}
+
+	return framework.LeaseExtend(resp.TokenTTL, resp.TokenMaxTTL, b.System())(req, data)
+}
+
+func (b *backend) pathLoginUpdate(req *logical.Request, data *framework.FieldData) (*logical.Response, error) {
```

Comment:
I think it'd be really nice if they can either pass in as `selector` and `user_id` or if they can pass in just `user_id` with a selector prepended with e.g. a colon. That way if an organization can easily transmit a string to a service but harder to transmit two strings, they can just merge them together.